lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp

```diff
@@ -36,17 +41,71 @@ bool LegalGridAnalysis::applyOverrides() {
 }
 
 void LegalGridAnalysis::analysisImplementation() {
-  // Placeholder, needs to be implemented. Go through all the grid sizes and
-  // check if they are legal based on tensor type and device/chip attributes.
-  // For now result of analysis is maximum supported grid size.
-  //
+  // A first incomplete implementation of the LegalGridAnalysis.
+  // This implementation is a placeholder and is meant to just enable testing of
+  // other components.
+
+  // Get output tensor type.
+  // TODO: This ignores multiple outputs...?
   RankedTensorType tensorType =
       mlir::cast<RankedTensorType>(op->getResult(0).getType());
   LayoutAttr layout = mlir::cast<LayoutAttr>(tensorType.getEncoding());
-  llvm::ArrayRef<int64_t> tensorShape = tensorType.getShape();
 
-  analysisResult.push_back(layout.withGrid(
-      op->getContext(), tensorShape,
-      GridAttr::get(op->getContext(), analysisInput.maxGrid.getShape())));
+  // DRAM
+  // No grid is set since the tensor is not sharded.
+  // TODO: Is this a viable solution or should we have a grid?
+  LayoutAttr dram =
+      layout.withMemorySpace(op->getContext(), MemorySpace::DeviceDRAM);
+  analysisResult.push_back(dram);
+
+  // L1 Interleaved (same as above)
+  LayoutAttr l1Interleaved =
+      layout.withMemorySpace(op->getContext(), MemorySpace::DeviceL1);
+  analysisResult.push_back(l1Interleaved);
```

Comment:
Is there any attribute that makes this one interleaved compared to sharded L1s below?

Reply:
No, the only difference is that interleaved does not have a gird set. Should we make this explicit in LayoutAttr?

Reply:
@jnie-TT is adding flatbuffer support.  Once that's in I think we can just add a new layout attribute to capture this.  Or maybe we should specialize a TTNN layout class?  So far it's just 1 enum, so specializing might be overkill, metal backend can just assert that this enum is always set to `None` or something.

Reply:
Great, thanks! Yeah lets go with new layout attribute.